include/compiler.h

```diff
@@ -30,6 +30,15 @@ WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.  */
 #ifndef COMPILER_H
 #define COMPILER_H
 
+#if defined(__STDC_VERSION__ ) && __STDC_VERSION__  >= 201112L // C11
+# include <thread.h>
+# define THREAD_LOCAL thread_local
+#elseif defined(__GNUC__)
```

Comment:
Is anyone NOT using C11 at this point?

Reply:
No idea.  I didn't really look at your requirements.  Could be some military use case....

Do you want a global replace of `__thread` to `thread_local` and take it out of the compile header altogether?  Or do you want the `THREAD_LOCAL` macro?